library/memory_buffer_alloc.c

```diff
@@ -247,6 +249,8 @@ static void *buffer_alloc_calloc( size_t n, size_t size )
 
     if( n != 0 && len / n != size )
```

Comment:
@andresag01 In my opinion, we should treat 0 size allocation as NULL, and fix the code where it is treated different